File: topic_5/Python/7-alter.py
# Recebe um inteiro e os números até ele (com os negativos desses numeros)

def alter(n):
    lista = []
    aux = -1
    cont = 1
    while(cont <= n):
        lista.append(cont)
        lista.append(aux)
        cont += 1
        aux -= 1
    return lista

print(alter(1) == [1,-1]) #True
print(alter(2) == [1,-1,2,-2]) #True
print(alter(4) == [1,-1,2,-2,3,-3,4,-4]) #True

File: topic_5/Haskell/7-alter.hs
-- Recebe um inteiro e os números até ele (com os negativos desses numeros)

alter 1 = [1, -1]
alter n = alter (n - 1) ++ valores
    where
        valores = [n, -n]

main = do
    a <- readLn :: IO Int
    print $ alter a